### ISSUE TITLE
Steamlink will not pair with GFE latest Moonlight build and Steam link Build 815

### ISSUE THREAD

**Issue by willsor** [moonlight-stream/moonlight-qt#622]
**Describe the bug**
Moonlight 3.1.4/3.1.3/3.1.2 on Steam Link Hardware can no longer proceed past the Pairing screen when using geforce experience as the host. 

**Steps to reproduce**
Prepare USB stick as normal. Boot up steam link hardware. Choose to start Moonlight. It asks you which computer to connect to, you select the right one, you input the stated code on the host computer, and the steam link just goes black and restarts. Boots you back to the Moonlight/My-Pc/Add Computer page.

**Affected games**
N/A

**Other Moonlight clients**
Works fine on Android. Cant test on ios

**Moonlight settings (please complete the following information)**
N/A couldnt get this far

**Client PC details (please complete the following information)**
- Steam link hardware build 815

**Server PC details (please complete the following information)**
Windows 10 2004
RTX 2080 super
nvidia 466.77

**Moonlight Logs (please attach)**
I dont know how to get these on steamlink, sorry.

**Comment by woody452** [comment]
having the same problem here. have DDU wiped drives and GFE and re installed no difference, tried with my partners laptop. does not make any difference. entering the pin code resets the steam link.

**Comment by woody452** [comment]
ok i tried a few older moonlight versions all with the same result. im now downgrading GFE to an older version to try that.

**Comment by woody452** [comment]
DOWNGRADE YOUR LINK. i flashed back to build 807 just after the UI update on the link and now moonlight 3.14 connects to my pc no problems

**Comment by Forceflow** [comment]
Just FYI: Existing pairings (that were made before this bug occured) work fine.

**Comment by willsor** [comment]
That's interesting

**Comment by Jorys-Paulin** [comment]
I can confirm the issue happens only when pairing. I have retrieved Moonlight's log and uploaded it here: [moonlight.log](https://github.com/moonlight-stream/moonlight-qt/files/6821720/moonlight.log). Here's the end of the file:
```
00:00:21 - Qt Warning: Qt Concurrent has caught an exception thrown from a worker thread.
This is not supported, exceptions thrown in worker threads must be
caught before control returns to Qt Concurrent.
```

**Comment by Forceflow** [comment]
Is build 809 the first one that breaks this behavior? This is listed as a Qt-related change.

From the [Steam Link build log](https://support.steampowered.com/kb_article.php?ref=8293-PDVM-4434): 

> Changes in build 809
>     Fixed downloading files over HTTPS using Qt networking

**Comment by woody452** [comment]
> 
> That's interesting, I wonder if I could now upgrade to the latest build now I have downgraded to create the pair as woody suggested

YES, this is possible. i just tried it. my link is now on the latest build, and the already established link to my desktop, still works. i have not tried a new connection to a different pc, but suspect that still wont work.

i am however noticing a slight delay that i never had before. maybe a half a second. not enough to put off playing a game, but cause i can see my desktop screen from my couch i can see that my big screen is behind slightly.

**Comment by cgutman** [comment]
Fixed by c658daeac65d262c6d1d28c61d1d39fe5bf076be

I have updated the v3.1.4 Steam Link build on the [releases page](https://github.com/moonlight-stream/moonlight-qt/releases) with the fix. Updating to the fixed build (`Moonlight-SteamLink-3.1.4-r2.zip`) will make pairing work again.
